tests/test_multi_file_viewer.ts

```diff
@@ -0,0 +1,999 @@
+        expect(multi_file_viewer.vm.$data.scrollable_height).toEqual(
+            wrapper.vm.$data.height_of_view_file_in
+        );
+
+    });
+
+    test('When a file is added it becomes the active file/tab ', () => {
+        @Component({
+            template:  `<div>
+                          <p class="first-button" @click="open_file(1)"> Add Kiwi </p>
+                          <p class="second-button" @click="open_file(2)"> Add Lime </p>
+                          <p class="third-button" @click="open_file(3)"> Add Mango </p>
+                          <p class="fourth-button" @click="open_file(4)"> Add Nectarine </p>
+                          <div class="mvf-container">
+                            <multi_file_viewer
+                              ref="multi_file">
+                            </multi_file_viewer>
+                          </div>
+                        </div>`,
+            components: {
+                'multi_file_viewer': MultiFileViewer
+            }
+        })
+
+        class WrapperComponent extends Vue {
+
+            open_file(file_number: number) {
+                let mfv: MultiFileViewer = <MultiFileViewer> this.$refs.multi_file;
+                if (file_number === 1) {
+                    mfv.add_to_viewing(this.file1, this.file_1_content);
+                }
+                else if (file_number === 2) {
+                    mfv.add_to_viewing(this.file2, this.file_2_content);
+                }
+                else if (file_number === 3) {
+                    mfv.add_to_viewing(this.file3, this.file_3_content);
+                }
+                else {
+                    mfv.add_to_viewing(this.file4, this.file_4_content);
+                }
+            }
+
+            file1 = "Kiwi Header";
+
+            file2 = "Lime Header";
+
+            file3 = "Mango Header";
+
+            file4 = "Nectarine Header";
+
+            file_1_content = "Kiwi Body";
+
+            file_2_content = "Lime Body";
+
+            file_3_content = "Mango Body";
+
+            file_4_content = "Nectarine Body";
+        }
+
+        const wrapper = mount(WrapperComponent);
+        let multi_file_viewer = wrapper.find({ref: 'multi_file'});
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(-1);
+
+        let add_first_file_button = wrapper.find(".first-button");
+        add_first_file_button.trigger('click');
+
+        let view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(0);
+        expect(multi_file_viewer.vm.$data.file_names_and_content.size).toEqual(1);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(1);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_1_content);
+
+        let add_second_file_button = wrapper.find(".second-button");
+        add_second_file_button.trigger('click');
```

Comment:
You might be able to save yourself some code by getting rid of the buttons and just calling the add and remove file functions directly on the multi file viewer component?